pkg/linear-pools/contracts/silo-linear-pool/SiloLinearPoolRebalancer.sol

```diff
@@ -47,21 +47,21 @@ contract SiloLinearPoolRebalancer is LinearPoolRebalancer, SiloExchangeRateModel
     function _wrapTokens(uint256 amount) internal override {
         // @dev In order to receive a sharesToken that can gain interest false must be entered for collateralOnly
         // deposit however, we need to approve the silo where we will be depositing our tokens to.
+        // false is set for _collateralOnly input parameter, so that we can gain interest on our deposit
         _mainToken.safeApprove(address(_silo), amount);
         _silo.deposit(address(_mainToken), amount, false);
     }
 
     function _unwrapTokens(uint256 wrappedAmount) internal override {
         // Withdrawing into underlying (i.e. DAI, USDC, etc. instead of sDAI or sUSDC). Approvals are not necessary here
         // as the wrapped token is simply burnt
-        uint256 mainAmount = _getRequiredTokensToWrap(wrappedAmount) - 1;
+        uint256 mainAmount = _convertWrappedToMain(_silo, address(_mainToken), wrappedAmount);
         // Same way we round up requiredTokensToWrap, we need to round down the main amount,
         // to make sure we have enough tokens to unwrap.
         _silo.withdraw(address(_mainToken), mainAmount, false);
     }
 
     function _getRequiredTokensToWrap(uint256 wrappedAmount) internal view override returns (uint256) {
-        uint256 rate = _calculateExchangeValue(_shareToken);
-        return wrappedAmount.mulDown(rate) + 1;
+        return _convertWrappedToMain(_silo, address(_mainToken), wrappedAmount) + 1;
```

Comment:
Adding the comment used by other LP Rebalancers for context:
```suggestion
        // `_convertWrappedToMain` returns how many main tokens will be returned when unwrapping. Since there's fixed
        // point divisions and multiplications with rounding involved, this value might be off by one. We add one to
        // ensure the returned value will always be enough to get `wrappedAmount` when unwrapping. This might result in
        // some dust being left in the Rebalancer.
        return _convertWrappedToMain(_silo, address(_mainToken), wrappedAmount) + 1;
```